### DISCUSSION TITLE
A couple of questions about fonts (harfbuzz, dynamic font size)

### DISCUSSION THREAD

**Discussion by Cicorione** [speedata/publisher#628] [Q&A]
Hi @pgundlach 

I have a couple question about fonts and I hope you can have an answer for both!

First I do believe my **harfbuzz** settings are wrong:

```xml
 <entry level="error" msg="[page 2 line 360, data line 6] Can&#39;t use mode=&#34;harfbuzz&#34; on LoadFontfile without harfbuzz library"></entry>
```

Am I missing some package on Debian that I need to install?

The other question is about **reducing font proportionally based on the space availability**, I am pretty sure that you wrote about that on the documentation but I am unable to find that passage again! 🤦‍♂️

Here is my example:

![table-title](https://github.com/user-attachments/assets/442abbb7-6116-4bdc-b418-103a95e6d73f)

My tables still have rendering issue so just ignore that.

Thanks in advance... 🙏

**Comment by pgundlach** [comment]
It helps me if you could ask separate questions in separate issues...

How did you install the speedata Publisher? A debian package or a ZIP file from the download page? Or did you compile yourself?

The binary seems to be missing the harfbuzz library.

The other question: https://doc.speedata.de/publisher/en/commandreference/nobreak/ -> reduce = fontfit

Does that help?

**Comment by Cicorione** [comment]
Sorry Patrick, I won't repeat it again.

Regarding **Harfbuzz** I have the some issue on both the binary provided from your repo and the one I compile for `arm64`, although I believe I have the library installed:

```
gir1.2-harfbuzz-0.0/stable,now 6.0.0+dfsg-3 arm64 [installed,automatic]
libharfbuzz-dev/stable,now 6.0.0+dfsg-3 arm64 [installed]
libharfbuzz-gobject0/stable,now 6.0.0+dfsg-3 arm64 [installed,automatic]
libharfbuzz-icu0/stable,now 6.0.0+dfsg-3 arm64 [installed,automatic]
libharfbuzz-subset0/stable,now 6.0.0+dfsg-3 arm64 [installed,automatic]
libharfbuzz0b/stable,now 6.0.0+dfsg-3 arm64 [installed,automatic]
```

**Comment by Cicorione** [comment]
Regarding the other question, that was exactly what I was looking for but it is not working:

```xml
<Paragraph textformat="nohyphen">
  <B>
    <NoBreak reduce="fontfit" factor="0.5">
      <Value select="upper-case(@title)" />
    </NoBreak>
  </B>
</Paragraph>
```

Maybe `nohyphen` is creating issue... 🤔

**Comment by pgundlach** [comment]
You have hit a bug with nobreak.... see #630 

Can you copy the full output of `sdluatex --version`?

You don't need an external harfbuzz library, it is included in the binary.

**Reply by Cicorione** [reply]
Good, the more bugs are spotted out the best Publisher becomes!

**Comment by pgundlach** [comment]
~You are using the arm64 binary? Did you compile it yourself?~

I forgot that I provide the arm64 binaries. I'd still be interested in the output of `sdluatex --version`

**Reply by Cicorione** [reply]
Hi Patrick, I remembered it inaccurately.
I tested my layout with the binary from your repo and I don't have any issue with harfbuzz. Thus I have issues with the version that I compile by myself, but if you're providing an `arm64` version I am going to use the one from your repository; however on FreeBSD I still need to compile Publisher so I would like to know which flag I have to use to compile or to link the binary to harfbuzz.

Thanks... 🙏

**Reply by pgundlach** [reply]
The output of `sdluatex --version` is still important to find the error. I'd assume it does not print `luahbtex`...

**Reply by Cicorione** [reply]
Hi Patrick, on `Debian arm64` and `FreeBSD amd64` I must create a symlink to `luatex` to compile Publisher.
Below the output from `Debiam arm64`:

```
ls -l /usr/bin/sdluatex
lrwxrwxrwx 1 root root 15 Oct  8 15:28 /usr/bin/sdluatex -> /usr/bin/luatex*

sdluatex --version
This is LuaTeX, Version 1.15.0 (TeX Live 2022/Debian)
Development id: 7509

Execute  'luatex --credits'  for credits and version details.

There is NO warranty. Redistribution of this software is covered by
the terms of the GNU General Public License, version 2 or (at your option)
any later version. For more information about these matters, see the file
named COPYING and the LuaTeX source.

LuaTeX is Copyright 2022 Taco Hoekwater and the LuaTeX Team.
```

**Reply by pgundlach** [reply]
This is the wrong binary. It must be luahbtex (not luatex). Then it should work. Do you compile the luatex binary yourself?

**Reply by Cicorione** [reply]
No I don't, I  use a symbolic link to luatex since the sdluatex available from the git repository it is only for `amd64` and doesn't work on `arm64` as well as on `FreeBSD`:

```
file publisher/luatex/linux/sdluatex 
publisher/luatex/linux/sdluatex: ELF 64-bit LSB executable, x86-64, version 1 (SYSV), dynamically linked, interpreter /lib64/ld-linux-x86-64.so.2, BuildID[sha1]=c6afff0ad32bf0b9eab8e0a3a2924210b4bde672, for GNU/Linux 2.6.32, stripped
```

**Reply by pgundlach** [reply]
Then you must link to luahbtex instead of luatex.

**Reply by Cicorione** [reply]
Using the right symlink resolved the issue:

```
sdluatex --version
This is LuaHBTeX, Version 1.15.0 (TeX Live 2022/Debian)
Development id: 7509

Execute  'luahbtex --credits'  for credits and version details.

There is NO warranty. Redistribution of this software is covered by
the terms of the GNU General Public License, version 2 or (at your option)
any later version. For more information about these matters, see the file
named COPYING and the LuaTeX source.

LuaTeX is Copyright 2022 Taco Hoekwater and the LuaTeX Team.
```

I look forward to test it on FreeBSD as well!

**Comment by Cicorione** [comment]
I am sorry to add that even this option:
```xml
<NoBreak reduce="fontsize" factor="0.7">
```

Did not work out... ☹️

**Reply by pgundlach** [reply]
This is #629
